app/src/main/java/com/github/fribourgsdp/radio/UserProfileActivity.kt

```diff
@@ -75,11 +79,18 @@ class UserProfileActivity : AppCompatActivity(), PlaylistAdapter.OnPlaylistClick
             text = if (user.spotifyLinked) "linked" else "unlinked"
         }
 
+        if (intent.getBooleanExtra(COMING_FROM_ADD_PLAYLIST_ACTIVITY_FLAG, false)){
+            addManualPlaylistToUser()
+            user.save(this)
+        }
+
```

Comment:
I think this is a pretty ugly way of doing this. I didn't realise this was how @nathanduchesne had implemented it for the spotify import. I think it makes much more sense to save the playlists directly in AddPlaylistsActivity by loading the user (works in any activity), changing its playlists, and saving the user (works in any activity). This completely bypasses the heavy Intent and this ugly check of origin activity. It also makes more sense for UserProfileActivity to just stupidly load the userProfile everytime and not have to handle special events. I will need to do a PR for switching to a single serializable implementation so I will probably remove the other instance of saving playlists through intents in that PR

Reply:
Okay, I'm going to refactor this. I will also modify the part from ImportSpotifyPlaylists, it might significantly simplify the intents.